### PR TITLE
CP-4493 app banner ist not triggered

### PR DESCRIPTION
**Pull request by jprajapati1996**

### CODE REVIEW

File: cleverpush/src/main/java/com/cleverpush/util/VersionComparatorNew.java [hunk 1]
```diff
@@ -0,0 +1,143 @@
+package com.cleverpush.util;
+
+import java.util.Comparator;
+
+public class VersionComparatorNew implements Comparator {
```
Comment (gbhrdt) [inline cleverpush/src/main/java/com/cleverpush/util/VersionComparatorNew.java:5]: Can we rename to `VersionComparator` ?

File: cleverpush/src/main/java/com/cleverpush/banner/AppBannerModule.java [hunk 2]
```diff
@@ -327,85 +328,86 @@ boolean isBannerTargetingAllowed(Banner banner) {
         if (allowed && banner.getAttributes() != null && banner.getAttributes().size() > 0) {
             allowed = false;
             for (HashMap<String, String> attribute : banner.getAttributes()) {
-                String attributeId = attribute.get("id");
-                String compareAttributeValue = attribute.get("value");
-                String relationString = attribute.get("relation");
+                Double attributeId = Double.valueOf(attribute.get("id"));
```
Comment (gbhrdt) [inline cleverpush/src/main/java/com/cleverpush/banner/AppBannerModule.java:331]: Revert changes

File: cleverpush/src/main/java/com/cleverpush/banner/AppBannerModule.java [hunk 3]
```diff
@@ -273,9 +275,8 @@ boolean isBannerTargetingAllowed(Banner banner) {
         if (banner == null) {
             return false;
         }
-
+
```
Comment (gbhrdt) [inline cleverpush/src/main/java/com/cleverpush/banner/AppBannerModule.java:278]: Revert changes

File: cleverpush/src/main/java/com/cleverpush/banner/AppBannerModule.java [hunk 4]
```diff
@@ -65,6 +66,7 @@ public class AppBannerModule {
     private final SharedPreferences sharedPreferences;
     private final SharedPreferences.Editor editor;
 
+
```
Comment (gbhrdt) [inline cleverpush/src/main/java/com/cleverpush/banner/AppBannerModule.java:69]: Remove empty line

**Review by gbhrdt — CHANGES_REQUESTED** [review]

File: cleverpush/src/main/java/com/cleverpush/util/VersionComparatorNew.java [hunk 5]
```diff
@@ -0,0 +1,143 @@
...
+        private final String _versionString;
+        private final int _length;
+
+        private int _position;
+        private int _number;
+        private String _suffix;
+        private boolean _hasValue;
+
+        VersionTokenizer(String versionString) {
+            if (versionString == null)
```
Comment (gbhrdt) [inline cleverpush/src/main/java/com/cleverpush/util/VersionComparatorNew.java:91]: Add { and }

File: cleverpush/src/main/java/com/cleverpush/util/VersionComparatorNew.java [hunk 6]
```diff
@@ -0,0 +1,143 @@
...
+            }
+            if (number1 > number2) {
+                // Number one is greater than number two
+                return 1;
+            }
+
+            boolean empty1 = suffix1.length() == 0;
+            boolean empty2 = suffix2.length() == 0;
+
+            if (empty1 && empty2) continue; // No suffixes
```
Comment (gbhrdt) [inline cleverpush/src/main/java/com/cleverpush/util/VersionComparatorNew.java:54]: Code style: Add { and }

**Review by gbhrdt — CHANGES_REQUESTED** [review]

File: cleverpush/src/main/java/com/cleverpush/util/VersionComparatorNew.java (hunk 1 above)
Comment (jprajapati1996) [reply cleverpush/src/main/java/com/cleverpush/util/VersionComparatorNew.java:5]: Yes sure

File: cleverpush/src/main/java/com/cleverpush/banner/AppBannerModule.java (hunk 2 above)
Comment (jprajapati1996) [reply cleverpush/src/main/java/com/cleverpush/banner/AppBannerModule.java:331]: Yes i already changes but not commited may be so i will Double to String converted

File: cleverpush/src/main/java/com/cleverpush/util/VersionComparatorNew.java (hunk 5 above)
Comment (jprajapati1996) [reply cleverpush/src/main/java/com/cleverpush/util/VersionComparatorNew.java:91]: VersionTokenizer(String versionString) {
            if (versionString == null){
                throw new IllegalArgumentException("versionString is null");
            }
            _versionString = versionString;
            _length = versionString.length();
        }

Comment (jprajapati1996) [reply cleverpush/src/main/java/com/cleverpush/util/VersionComparatorNew.java:91]: is it okay

File: cleverpush/src/main/java/com/cleverpush/util/VersionComparatorNew.java (hunk 6 above)
Comment (jprajapati1996) [reply cleverpush/src/main/java/com/cleverpush/util/VersionComparatorNew.java:54]: yes i changed

File: cleverpush/src/main/java/com/cleverpush/banner/AppBannerModule.java (hunk 3 above)
Comment (jprajapati1996) [reply cleverpush/src/main/java/com/cleverpush/banner/AppBannerModule.java:278]: Done

File: cleverpush/src/main/java/com/cleverpush/banner/AppBannerModule.java (hunk 4 above)
Comment (jprajapati1996) [reply cleverpush/src/main/java/com/cleverpush/banner/AppBannerModule.java:69]: Done

**Pull request merged**